libs/web/src/hooks/useWalletTransactions.ts

```
@@ -227,6 +303,386 @@ const calculateAssetAmounts = (
+  const amount0In = safeStringToBigInt(action.amount0In);
+  const amount0Out = safeStringToBigInt(action.amount0Out);
+  const amount1In = safeStringToBigInt(action.amount1In);
+  const amount1Out = safeStringToBigInt(action.amount1Out);
+
+  switch (action.type) {
+    case "SWAP":
+      // For swaps, we should have either:
+      // - asset0 in and asset1 out (asset0 -> asset1)
+      // - asset1 in and asset0 out (asset1 -> asset0)
+      // At least one input and one output should be non-zero
+      const hasInput = amount0In > BigInt(0) || amount1In > BigInt(0);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)